### ISSUE TITLE
Menu - Accessibility violation in the 'with icon' story

### ISSUE THREAD

**Issue by sianford** [Sage/carbon#6175]
### Current behaviour

There is an accessibility violation for reason "Elements must only use allowed ARIA attributes" in the 'with icon' story.

<img width="1271" alt="Screenshot 2023-07-05 at 16 44 45" src="https://github.com/Sage/carbon/assets/12828099/8f7db2a3-c180-4c27-add6-ce28bd73f588">

### Expected behaviour

There are no accessibility violations in the Menu component.

### CodeSandbox or Storybook URL

https://carbon.sage.com/?path=/story/menu--with-icon-story

### JIRA Ticket (Sage Only)

FE-6048

### Suggested Solution

Resolve violation.

### Carbon Version

119.5.0

### What browsers are you seeing the problem on?

Firefox, Chrome, Safari, Microsoft Edge

### What Operating System are you seeing the problem on?

MacOS

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.

**Comment by robinzigmond** [comment]
FE-6055
